include/zephyr/bluetooth/mesh/access.h

```diff
@@ -904,18 +916,16 @@ struct bt_mesh_model {
 	/** Model callback structure. */
 	const struct bt_mesh_model_cb * const cb;
 
-#ifdef CONFIG_BT_MESH_MODEL_EXTENSIONS
-	/* Pointer to the next model in a model extension list. */
-	struct bt_mesh_model *next;
-#endif
-
 #ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
 	/* Pointer to the array of model metadata entries. */
-	struct bt_mesh_models_metadata_entry **metadata;
+	struct bt_mesh_models_metadata_entry ** const metadata;
 #endif
 
+	/** Model ctx variables. */
+	struct bt_mesh_model_ctx * const ctx;
+
 	/** Model-specific user data */
-	void *user_data;
+	void * const user_data;
```

Comment:
Can we take this opportunity to rename `user_data` to `modelinst` or `modelptr`? After making this user_data as 'const', makes is even less 'user' specific thing.